treeherder/etl/seta.py

```diff
@@ -0,0 +1,210 @@
+
+    def _testtype_job_identifier(self, job):
+        '''
+        job - treeherder job
+        '''
+        job_identifier = None
+        testtype = None
+
+        # the testtype of builbot job can been found in 'ref_data_name'
+        # like web-platform-tests-4 in "Ubuntu VM 12.04 x64 mozilla-inbound
+        #  opt test web-platform-tests-4"
+        if job['build_system_type'] == 'buildbot':
+            testtype = job['ref_data_name'].split(' ')[-1]
+            job_identifier = job['ref_data_name']
+
+        # taskcluster's testtype is a part of its 'job_type_name' like reftest-2
+        # for [TC] Linux64 reftest-2
+        elif job['build_system_type'] == 'taskcluster':
+            # The test name on TaskCluster comes to a sort of combination
+            # (e.g desktop-test-linux64/debug-jittests-3) and asan job can
+            # been referenced as a opt job.
+            # NOTE: jobs like 'MacOSX64 Static Analysis Opt' will be left aside
+            # because they are not a test job.
+            # NOTE: The day we have Buildbot bridge in-tree we will have to support it
+            if job['ref_data_name'].startswith('desktop-test') or \
+               job['ref_data_name'].startswith('android-test'):
+
+                # we want the build type(debug or opt) to separate the job_type_name
+                # (e.g desktop-test-linux64/debug-jittests-3)
+                separator = job['platform_option'] \
+                    if job['platform_option'] != 'asan' else 'opt'
+                # we should get "jittests-3" as testtype for job_type_name like
+                # desktop-test-linux64/debug-jittests-3
+                testtype = job['job_type_name'].split(
+                    '{buildtype}-'.format(buildtype=separator))[-1]
+                job_identifier = job['ref_data_name']
+
+        return testtype, job_identifier
+
+    def query_jobtypes(self):
+        """Query all available jobtypes and return it as list"""
+        return self.jobtypes
+
+    def query_jobnames(self):
+        """Query all jobnames including buildtype and groupcode, then return them as list"""
+        return self.jobnames
+
+    def _get_jobnames(self, job, testtype, job_identifier):
+        signature = ''  # TH specific
+        buildplatform = job['build_system_type']
+        buildtype = job['platform_option']
+        platform = job['platform']
+        name = testtype
+        jobname = ''  # TH specific
+        job_identifier = job_identifier
+        group = _get_group(name)
+        groupcode = _get_group_code(self.tbplnames, name)
+        code = _get_code(self.tbplnames, name)
+        data = {'buildplatform': buildplatform, 'jobname': jobname, 'signature': signature,
+                'job_type_name': group, 'job_group_symbol': groupcode,
+                'name': name, 'job_type_symbol': code,
+                'ref_data_name': job_identifier,
+                'platform': platform, 'buildtype': buildtype}
```

Comment:
the group/groupcode/code/jobname/job_identifier are used just for the UI to display which jobs we run/hide by default.  I am not sure that is needed once inside of TC- possibly a bug to investigate a TH specific UI and add a comment here are link to this code so we can cleanup as necessary.